plugins/chat/lib/chat/channel_archive_service.rb

```diff
@@ -107,22 +108,119 @@ def execute
         # Another future improvement is to send a MessageBus message for each
         # completed batch, so the UI can receive updates and show a progress
         # bar or something similar.
+
+        def create_post_from_batch(chat_messages, batch_thread_ranges)
+          create_post(
+            Chat::TranscriptService.new(
+              chat_channel,
+              chat_channel_archive.archived_by,
+              messages_or_ids: chat_messages,
+              thread_ranges: batch_thread_ranges,
+              opts: {
+                no_link: true,
+                include_reactions: true,
+              },
+            ).generate_markdown,
+          ) { delete_message_batch(chat_messages.map(&:id)) }
+        end
+
+        buffer = []
+        thread_ranges = {}
+        batch_thread_ranges = {}
+
         chat_channel
           .chat_messages
-          .find_in_batches(batch_size: ARCHIVED_MESSAGES_PER_POST) do |chat_messages|
-            create_post(
-              Chat::TranscriptService.new(
-                chat_channel,
-                chat_channel_archive.archived_by,
-                messages_or_ids: chat_messages,
-                opts: {
-                  no_link: true,
-                  include_reactions: true,
-                },
-              ).generate_markdown,
-            ) { delete_message_batch(chat_messages.map(&:id)) }
+          .order("created_at ASC")
+          .find_in_batches(batch_size: ARCHIVED_MESSAGES_PER_POST) do |message_batch|
+            thread_ids = message_batch.map(&:thread_id).compact.uniq
+            threads =
+              chat_channel.chat_messages.where(thread_id: thread_ids).order("created_at ASC").to_a
+
```

Comment:
This check is not enough, you can have a thread_id and actually no replies, we need to be sure there's at least one reply otherwise we shouldnt consider it a thread.

If you want to repro this, just hit reply on a message in a channel and don't write anything in the thread panel.